packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol

```diff
@@ -113,50 +114,66 @@ contract EthERC20Bridge is L1Buddy {
     function updateTokenInfo(
         address erc20,
         uint256 maxGas,
+        uint256 gasPriceBid,
+        bool isERC20
+    ) external payable onlyIfConnected {
+        string memory name = SafeERC20Namer.tokenName(erc20);
+        string memory symbol = SafeERC20Namer.tokenSymbol(erc20);
+        uint8 decimals = ERC20(erc20).decimals();
+
+        bytes4 _selector = isERC20
+            ? ArbTokenBridge.updateERC777TokenInfo.selector
+            : ArbTokenBridge.updateERC20TokenInfo.selector;
+
+        sendPairedContractTransaction(
+            maxGas,
+            gasPriceBid,
+            abi.encodeWithSelector(_selector, erc20, name, symbol, decimals)
+        );
+    }
+
+    function depositAsERC777(
+        address erc20,
```

Comment:
maybe rename parameter to erc777 ?

Reply:
I don't think so, because the user is depositing an ERC20, not an ERC777